### PR TITLE
DOC: Fix latexpdf build

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is a rebase of #15797, plus additional fixes for new errors and a patch for macOS font paths.

## PR Checklist

- [n/a] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: tutorials/text/mathtext.py [hunk 1]
```diff
@@ -94,7 +94,7 @@
 
 .. math::
 
-    \frac{3}{4} \binom{3}{4} \genfrac{}{}{0}{}{3}{4}
+    \frac{3}{4} \binom{3}{4} \genfrac{}{}{0pt}{}{3}{4}
```
Comment (QuLogic) [inline tutorials/text/mathtext.py:97]: This dimension appears to be required by LaTeX, but not our parser. I couldn't find anything in the parser that could require it, (if we wanted to do that)?

cc @anntzer

Comment (anntzer) [reply tutorials/text/mathtext.py:97]: Indeed, `\genfrac` behaves differently between mathtext and TeX: TeX wants (quite reasonably) a unit whereas we have not implemented support for it.  I don't actually know what's the unit we implicitly use?
I guess adding support for units there is not "too" hard (mostly just figuring out the right multiplier); but in the meantime I guess you could just exclude that from the pdf build using `.. only::` or `.. ifconfig::`?

Comment (QuLogic) [reply tutorials/text/mathtext.py:97]: Well this is in `.. math::`; according to the build here, it works fine with the unit in regular HTML builds, too. It's just inconsistent with the Matplotlib version in the plain text source.

Comment (anntzer) [reply tutorials/text/mathtext.py:97]: I guess you should mostly just open an issue for supporting `{0pt}` on our side?  (which seems like a reasonable request)  Or did I miss something?

Comment (QuLogic) [reply tutorials/text/mathtext.py:97]: Yes, as it seems I didn't miss anything, #19001.

**Comment by tacaswell** [comment]
This is failing for me on arch with

```
kpathsea: Running mktextfm XITS-Regular
/usr/share/texmf-dist/web2c/mktexnam: Could not map source abbreviation X for XITS-Regular.
/usr/share/texmf-dist/web2c/mktexnam: Need to update /usr/share/texmf-dist/fonts/map/fontname/special.map?
mktextfm: Running mf-nowin -progname=mf \mode:=ljfour; mag:=1; ; nonstopmode; input XITS-Regular
This is METAFONT, Version 2.7182818 (TeX Live 2020/Arch Linux) (preloaded base=mf)


kpathsea: Running mktexmf XITS-Regular
! I can't find file `XITS-Regular'.
<*> ...; mag:=1; ; nonstopmode; input XITS-Regular
                                                  
Please type another input file name
! Emergency stop.
<*> ...; mag:=1; ; nonstopmode; input XITS-Regular
                                                  
Transcript written on mfput.log.
grep: XITS-Regular.log: No such file or directory
mktextfm: `mf-nowin -progname=mf \mode:=ljfour; mag:=1; ; nonstopmode; input XITS-Regular' failed to make XITS-Regular.tfm.
kpathsea: Appending font creation commands to missfont.log.


! Package fontspec Error: The font "XITS-Regular" cannot be found.

For immediate help type H <return>.
 ...                                              
                                                  
l.31 ]
      
```

I do have the XITS-Regular file on my system:

```
jupiter@12:53  ➤  locate XITS-Regular
/usr/share/texmf-dist/fonts/opentype/public/xits/XITS-Regular.otf
```

If I go through the darwin code path then it builds cleanly  so I think we need to be a bit more precise about sorting out how to specify the fonts than just if we are on darwin :(.

**Comment by QuLogic** [comment]
That's unfortunate; I was hoping to avoid figuring out how to query `kpathsea`.

**Comment by anntzer** [comment]
Not really sure what you exactly need here but is it something along the lines of `\setmathfont{XITS-Regular.otf}[Path=/usr/share/texmf-dist/fonts/opentype/public/xits/]` where the full path is basically given to you by `kpsewhich XITS-Regular.otf`?  (Feel free to ignore if irrelevant...)

**Comment by QuLogic** [comment]
Maybe. @tacaswell what is the output of `kpsewhich XITS-Regular.otf` on Arch?

**Comment by tacaswell** [comment]
on arch:

```
~                                                                                                                                                                    
 11:36:16 $  kpsewhich XITS-Regular.otf                                                                                                                              
/usr/share/texmf-dist/fonts/opentype/public/xits/XITS-Regular.otf
```

on ubuntu

```
tcaswell@sharon:~$ kpsewhich XITS-Regular.otf 
/usr/share/texlive/texmf-dist/fonts/opentype/public/xits/XITS-Regular.otf
```

**Comment by QuLogic** [comment]
I spent a while trying to get this to work with `kpsewhich`, but then I looked back at the `fontspec` docs and found that it had some builtin support to do this. This currently works on Fedora and macOS, and hopefully works on Arch since it should trigger the fallback like it does on macOS.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
I tested that this now builds cleanly for me on an Arch box.

**Comment by tacaswell** [comment]
Merged on a single review because this configuration to build the latex docs and some of these commits came from the OP.

@jfbu Thank you for getting this work started!

**Comment by jfbu** [comment]
@tacaswell @QuLogic great to hear that you seem to have solved tricky multi-platform issues there!

**Pull request merged**